### ISSUE TITLE
Rivera Dancer ICD bugs are not correctly simulated

### ISSUE THREAD

**Issue by haasn** [d07RiV/d3planner#185]
On live servers, Rivera Dancer's increased attack speed does increase the attack speed of item procs like Gyana Na Kashu and Obsidian Ring of the Zodiac.

So for example, suppose you have an APS of exactly 1 (60 frames), then with the 50% buff from rivera dancers this would go up to 1.5 (40 frames), but the ICD on obsidian ring / gyana na kashu is still 60 frames, so the simulation looks like this:

t=0f: use LTK; gyana+obsidian proc with 60f ICD
t=40f: use LTK, gyana+obsidian still on ICD
t=80f: use LTK, gyana+obsidian proc
and so on

Basically only every other attack or so actually procs the fireball/obsidian ring. The same is also true for the copied LTK from rabid strike (although it doesn't seem like d3planner correctly simulates RS at all right now).

**Comment by haasn** [comment]
I'm actually not sure if my math/explanation is correct. In practice, it doesn't quite seem 1:2, for example 15 LTK usages = 9 fireballs according to Davlok's tests. I don't know the exact math, but we could do a measurement and get a good approximation for it instead.

**Comment by d07RiV** [comment]
I've tried to pinpoint OROZ mechanics, it seemed very inconsistent so I gave up on that and just made it give 1 proc per cast of anything. I'll see if I can find anything more specific on this.

I thought RS should work now?

**Comment by haasn** [comment]
Could be that the RS bug has been fixed since the date of the post I was sourcing that from. Don't know for certain. But at any rate, RS feels very weak compared to how strong it should be in theory (if it worked correctly and scaled off both rivera dancers and scarbringer).
